h/views/groups.py

```diff
@@ -149,25 +144,43 @@ def read_unauthenticated(group, request):
     return {'group': group}
 
 
-@view_config(route_name='group_read_noslug', request_method='GET')
-def read_noslug(group, request):
-    _check_slug(group, request)
+@view_defaults(route_name='group_read',
```

Comment:
Yep. I like to use the view config as much as possible to make Pyramid choose the right view, instead of having branching logic in Python. Makes the view functions simpler